app/src/main/java/ch/epfl/sweng/rps/persistence/Cache.kt

```diff
@@ -0,0 +1,79 @@
+package ch.epfl.sweng.rps.persistence
+
+import android.content.Context
+import android.util.Log
+import ch.epfl.sweng.rps.db.Env
+import ch.epfl.sweng.rps.db.FirebaseHelper
+import ch.epfl.sweng.rps.db.FirebaseReferences
+import ch.epfl.sweng.rps.db.FirebaseRepository
+import ch.epfl.sweng.rps.models.User
+import ch.epfl.sweng.rps.models.UserStat
+import ch.epfl.sweng.rps.services.ServiceLocator
+import java.net.InetAddress
+
+
+class Cache private constructor(private val ctx:Context, val preferFresh:Boolean = false) {
+    companion object {
+        var cache:Cache? = null
+        fun getInstance(preferFresh: Boolean = false): Cache? {
+            return cache
+        }
+        fun createInstance(ctx : Context ): Cache {
+            cache = Cache(ctx.applicationContext, true)
+            return cache!!
+        }
+    }
```

Comment:
The `preferFresh` constructor argument here is always true, and the `preferFresh` argument to `getInstance` is never used.

Reply:
Yes I know, they are here because I might want to use them in future, but I can remove them if you think it pollutes code too much.